storage-node/src/cache/policy/lru.rs

```diff
@@ -46,37 +46,60 @@ impl<K: DataStoreCacheKey, V: DataStoreCacheValue> LruCache<K, V> {
                 value.size(),
                 self.max_capacity
             );
-            return false;
+            return (false, None);
         }
         if let Some(cache_value) = self.cache_map.get(&key) {
             // If the key already exists, update the cache size.
             self.size -= cache_value.size();
         }
         self.size += value.size();
         self.cache_map.insert(key.clone(), value);
+        let mut evicted_keys = Vec::new();
         while self.size > self.max_capacity {
             if let Some((key, cache_value)) = self.cache_map.pop_front() {
                 println!("-------- Evicting Key: {:?} --------", key);
+                evicted_keys.push(key);
                 self.size -= cache_value.size();
             }
         }
-        true
+        if !evicted_keys.is_empty() {
+            (true, Some(evicted_keys))
+        } else {
+            (true, None)
+        }
     }
 
     fn peek_value(&self, key: &K) -> Option<&V> {
         self.cache_map.get(key)
     }
+
+    fn pop_value(&mut self, key: &K) -> Option<V> {
+        if let Some(cache_value) = self.cache_map.remove(key) {
+            self.size -= cache_value.size();
+            Some(cache_value)
+        } else {
+            None
+        }
+    }
 }
 
-impl DataStoreCache for LruCache<ParpulseDataStoreCacheKey, ParpulseDataStoreCacheValue> {
+impl DataStoreReplacer for LruReplacer<ParpulseDataStoreCacheKey, ParpulseDataStoreCacheValue> {
     fn get(&mut self, key: &ParpulseDataStoreCacheKey) -> Option<&ParpulseDataStoreCacheValue> {
         self.get_value(key)
     }
 
-    fn put(&mut self, key: ParpulseDataStoreCacheKey, value: ParpulseDataStoreCacheValue) -> bool {
+    fn put(
+        &mut self,
+        key: ParpulseDataStoreCacheKey,
+        value: ParpulseDataStoreCacheValue,
+    ) -> (bool, Option<Vec<ParpulseDataStoreCacheKey>>) {
         self.put_value(key, value)
     }
 
+    fn pop(&mut self, key: &ParpulseDataStoreCacheKey) -> Option<ParpulseDataStoreCacheValue> {
```

Comment:
What is the use case of explicitly calling pop()?

Reply:
I couldn't think of any use case right now and I have in fact been thinking of removing the unused interfaces from the trait...

Reply:
Oh I will remove them, it must be something I wrongly thought them useful before...